### ISSUE TITLE
Failing to build from flake

### ISSUE THREAD

**Issue by pupbrained** [VirtCode/hypr-dynamic-cursors#46]
Hi, I wanted to try this plugin out, but upon adding the plugin to my NixOS flake config and running a system rebuild the buildPhase fails with a lot of the same include error:

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/7c8wmcvpw9za7nca17k73yz59qmn85jp-5230bybiqijskxz4hc0vbydia5v8f5dk-source
source root is 5230bybiqijskxz4hc0vbydia5v8f5dk-source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/p6k7xp1lsfmbdd731mlglrdj2d66mr82-bash-5.2p37/bin/bash
mkdir -p out
g++ -shared -Wall --no-gnu-unique -fPIC ./src/cursor.cpp ./src/highres.cpp ./src/main.cpp ./src/renderer.cpp ./src/config/ShapeRule.cpp ./src/config/confi>
In file included from ./src/config/config.hpp:3,
                 from ./src/cursor.cpp:1:
/nix/store/83qyvl7mj6a0dxbpx9hmr6gl8iagjq5v-hyprlang-0.6.0+date=2024-12-16_16e59c1-dev/include/hyprlang.hpp:11:10: fatal error: print: No such file or dir>
   11 | #include <print>
      |          ^~~~~~~
compilation terminated.
In file included from /nix/store/vj0bh428v6k09008g9hmgxha6nk54d70-hyprland-0.46.0+date=2024-12-18_df06cb4-dev/include/hyprland/src/plugins/PluginAPI.hpp:3>
                 from ./src/globals.hpp:3,
                 from ./src/highres.cpp:1:
/nix/store/83qyvl7mj6a0dxbpx9hmr6gl8iagjq5v-hyprlang-0.6.0+date=2024-12-16_16e59c1-dev/include/hyprlang.hpp:11:10: fatal error: print: No such file or dir>
   11 | #include <print>
      |          ^~~~~~~
compilation terminated.
In file included from /nix/store/vj0bh428v6k09008g9hmgxha6nk54d70-hyprland-0.46.0+date=2024-12-18_df06cb4-dev/include/hyprland/src/plugins/PluginAPI.hpp:3>
                 from ./src/main.cpp:4:
/nix/store/83qyvl7mj6a0dxbpx9hmr6gl8iagjq5v-hyprlang-0.6.0+date=2024-12-16_16e59c1-dev/include/hyprlang.hpp:11:10: fatal error: print: No such file or dir>
   11 | #include <print>
      |          ^~~~~~~
compilation terminated.
In file included from /nix/store/vj0bh428v6k09008g9hmgxha6nk54d70-hyprland-0.46.0+date=2024-12-18_df06cb4-dev/include/hyprland/src/plugins/PluginAPI.hpp:3>
                 from ./src/globals.hpp:3,
                 from ./src/renderer.cpp:1:
/nix/store/83qyvl7mj6a0dxbpx9hmr6gl8iagjq5v-hyprlang-0.6.0+date=2024-12-16_16e59c1-dev/include/hyprlang.hpp:11:10: fatal error: print: No such file or dir>
   11 | #include <print>
      |          ^~~~~~~
compilation terminated.
In file included from ./src/config/ShapeRule.hpp:7,
                 from ./src/config/ShapeRule.cpp:1:
/nix/store/83qyvl7mj6a0dxbpx9hmr6gl8iagjq5v-hyprlang-0.6.0+date=2024-12-16_16e59c1-dev/include/hyprlang.hpp:11:10: fatal error: print: No such file or dir>
   11 | #include <print>
      |          ^~~~~~~
compilation terminated.
In file included from /nix/store/vj0bh428v6k09008g9hmgxha6nk54d70-hyprland-0.46.0+date=2024-12-18_df06cb4-dev/include/hyprland/src/plugins/PluginAPI.hpp:3>
                 from ./src/config/../globals.hpp:3,
                 from ./src/config/config.cpp:1:
/nix/store/83qyvl7mj6a0dxbpx9hmr6gl8iagjq5v-hyprlang-0.6.0+date=2024-12-16_16e59c1-dev/include/hyprlang.hpp:11:10: fatal error: print: No such file or dir>
   11 | #include <print>
      |          ^~~~~~~
compilation terminated.
In file included from ./src/mode/../config/config.hpp:3,
                 from ./src/mode/ModeRotate.cpp:1:
/nix/store/83qyvl7mj6a0dxbpx9hmr6gl8iagjq5v-hyprlang-0.6.0+date=2024-12-16_16e59c1-dev/include/hyprlang.hpp:11:10: fatal error: print: No such file or dir>
   11 | #include <print>
      |          ^~~~~~~
compilation terminated.
In file included from ./src/mode/../config/config.hpp:3,
                 from ./src/mode/ModeStretch.cpp:3:
/nix/store/83qyvl7mj6a0dxbpx9hmr6gl8iagjq5v-hyprlang-0.6.0+date=2024-12-16_16e59c1-dev/include/hyprlang.hpp:11:10: fatal error: print: No such file or dir>
   11 | #include <print>
      |          ^~~~~~~
compilation terminated.
In file included from ./src/mode/../config/config.hpp:3,
                 from ./src/mode/ModeTilt.cpp:3:
/nix/store/83qyvl7mj6a0dxbpx9hmr6gl8iagjq5v-hyprlang-0.6.0+date=2024-12-16_16e59c1-dev/include/hyprlang.hpp:11:10: fatal error: print: No such file or dir>
   11 | #include <print>
      |          ^~~~~~~
compilation terminated.
In file included from /nix/store/vj0bh428v6k09008g9hmgxha6nk54d70-hyprland-0.46.0+date=2024-12-18_df06cb4-dev/include/hyprland/src/plugins/PluginAPI.hpp:3>
                 from ./src/other/../globals.hpp:3,
                 from ./src/other/Shake.cpp:1:
/nix/store/83qyvl7mj6a0dxbpx9hmr6gl8iagjq5v-hyprlang-0.6.0+date=2024-12-16_16e59c1-dev/include/hyprlang.hpp:11:10: fatal error: print: No such file or dir>
   11 | #include <print>
      |          ^~~~~~~
compilation terminated.
make: *** [Makefile:8: dynamic-cursors.so] Error 1
```

I assume it's just missing a build input or something (if this isn't an issue on my end) but any help is appreciated.

**Comment by VirtCode** [comment]
hmm this looks really weird, can't seem to reproduce it on my end, but I am also not using nix

can you share the relevant parts of your nix config?

**Comment by pupbrained** [comment]
> hmm this looks really weird, can't seem to reproduce it on my end, but I am also not using nix
> 
> can you share the relevant parts of your nix config?

All I did was add the lines as specified in the README - but I did find [this issue](https://github.com/hyprwm/hyprlang/issues/60) on the hyprlang repo which has the same errors. The only thing is - that's already been closed and fixed, so I don't know why the error is now occurring here. (For reference, I'm on the latest commit of the hyprland repo/flake and everything else related to it builds fine.)

**Comment by VirtCode** [comment]
maybe @bloxx12 has more of an idea what could be causing this, I really don't have much clue about nix stuff

**Comment by bloxx12** [comment]
Will take a look shortly.

**Comment by pupbrained** [comment]
> @VirtCode I believe the problem is that hyprlang started using gcc14, but in this flake the version is still gcc13

oh wow, good catch - changing it to gcc14 fixes the build on my end.

**Comment by bloxx12** [comment]
Can reproduce, changing to gcc14 fixes it. Will make a pr.

**Comment by VirtCode** [comment]
merged #47, should be fixed then i guess
thanks yall
